openedx/features/content_type_gating/partitions.py

```diff
@@ -73,7 +78,91 @@ def create_content_gating_partition(course):
 
 
 class ContentTypeGatingPartition(UserPartition):
-    pass
+
+    def access_denied_fragment(self, block, user, user_group, allowed_groups):
+        ecomm_service = EcommerceService()
+        ecommerce_checkout = ecomm_service.is_enabled(user)
+        ecommerce_checkout_link = ''
+        ecommerce_bulk_checkout_link = ''
+        verified_mode = None
+        CourseMode = apps.get_model('course_modes.CourseMode')
+        modes = CourseMode.modes_for_course_dict(block.scope_ids.usage_id.course_key)
+        verified_mode = modes.get(CourseMode.VERIFIED, '')
+
+        if ecommerce_checkout and verified_mode.sku:
+            ecommerce_checkout_link = ecomm_service.get_checkout_page_url(verified_mode.sku)
+
+        if verified_mode is None:
+            return None
+
+        request = crum.get_current_request()
```

Comment:
we've been using CONTENT_TYPE_GATING_FLAG.is_enabled() which you can see on master

Reply:
This isn't actually checking the waffle flag, it's checking the user agent to detect mobile.